eZ/Publish/Core/IO/IOMetadataHandler/LegacyDFSCluster.php

```diff
@@ -291,4 +283,37 @@ protected function mapSPIBinaryFileCreateStructToSPIBinaryFile(SPIBinaryFileCrea
 
         return $spiBinaryFile;
     }
+
+    /**
+     * Generate the correct SQL query to insert an entry.
+     *
+     * @return string
+     */
+    protected function getFileInsertQuery()
+    {
+        $dbDriver = $this->db->getDriver();
+
+        if ($dbDriver instanceof Connection && $dbDriver->getName() === 'pdo_mysql') {
```

Comment:
I'm sorry, but I cannot accept this solution. This clearly requires a strategy. If there's a different query for different DBMS you need to create a Gateway for each one and some registry which will hold different implementations per db driver (or maybe rather by `$this->db->getDatabasePlatform()->getName()`).

To avoid complicating the solution for `6.7` you could skip explicit registry part and inject (in Symfony DIC) a simple associative array of the form e.g.
``` php
[
    'mysql' => MySQLGateway
    'ansi' => AnsiSQLGateway
]
```
and make  
``` php 
private function getGateway(string $databasePlatformName)
```
method which will return proper gateway if exists, if not, an ANSI one. This of course raises a natural question - what if ansi was not injected? There should be a `\RuntimeException` in this case.

Might feel like an overkill, but if we say we're SOLID, we should always be.

Reply:
@alongosz All fair points, and kind of expected. Just noting that testing & refactoring will delay this a good deal. I'll see if I can hand that part off to Engineering, and give the fix as is to the client for their testing (with the appropriate warnings that it's not prod-ready).